File: app/src/main/java/com/lamnt/foodorder/view/fragment/base/BaseDialogFragment.java
package com.lamnt.foodorder.view.fragment.base;

import android.app.Activity;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import java.util.Objects;

import butterknife.ButterKnife;

public abstract class BaseDialogFragment extends DialogFragment {
    protected @LayoutRes
    int layoutRes;
    protected Activity mActivity;

    @Override
    public void onStart() {
        super.onStart();
        try {
            WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
            lp.copyFrom(Objects.requireNonNull(this.getDialog().getWindow()).getAttributes());
            lp.width = WindowManager.LayoutParams.MATCH_PARENT;
            lp.height = WindowManager.LayoutParams.WRAP_CONTENT;
            lp.gravity = Gravity.CENTER;
            this.getDialog().setCancelable(false);
            this.getDialog().getWindow().setAttributes(lp);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mActivity = getActivity();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        mActivity = getActivity();
        View mView = inflater.inflate(layoutRes, container, false);
        ButterKnife.bind(this, mView);
        unit(mView);
        return mView;
    }

    protected abstract void unit(View v);
}

File: app/src/main/java/com/lamnt/foodorder/view/base/BaseDialogFragment.kt
package com.lamnt.foodorder.view.base

import android.os.Bundle
import android.view.*
import androidx.annotation.LayoutRes
import androidx.fragment.app.DialogFragment

abstract class BaseDialogFragment : DialogFragment() , View.OnClickListener{
    @JvmField
    @LayoutRes
    protected var layoutRes = 0
    private var views : List<View> = ArrayList()


    abstract fun setViewOnClick(): List<View>
    abstract fun onViewClicked(id : Int)

    override fun onStart() {
        super.onStart()
        try {
            val lp = WindowManager.LayoutParams()
            lp.copyFrom(this.dialog!!.window?.attributes)
            lp.width = WindowManager.LayoutParams.MATCH_PARENT
            lp.height = WindowManager.LayoutParams.WRAP_CONTENT
            lp.gravity = Gravity.CENTER
            this.dialog!!.setCancelable(false)
            this.dialog!!.window!!.attributes = lp
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(layoutRes, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        unit()
        views = setViewOnClick()
        registerListener(views)
    }

    abstract fun unit()

    private fun registerListener(views : List<View>){
        for (view in views){
            view.setOnClickListener(this@BaseDialogFragment)
        }
    }
    override fun onClick(v: View?) {
        onViewClicked(v!!.id)
    }
}